### PR TITLE
Feature: support Sierra 1.6

### PR DESCRIPTION
**Pull request by odesenfans**
Problem: we need the ability to compile Sierra 1.6 contracts to prove some blocks

Solution: bump all the dependencies to be able to use the Rust cairo crates v2.7.x.

* Bump cairo-vm to 1.0.1
* Bump Blockifier to 0.8.0-rc.2
* Bump the cairo crates to 2.7.1.

Issue Number: N/A

## Type

- [x] feature
- [ ] bugfix
- [x] dev (no functional changes, no API changes)
- [ ] fmt (formatting, renaming)
- [ ] build
- [ ] docs
- [ ] testing

## Breaking changes?

- [x] yes
- [ ] no

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -58,7 +58,7 @@ serde_json = { version = "1.0.105", features = ["arbitrary_precision"] }
 serde_with = "3.3.0"
 serde_yaml = "0.9.25"
 starknet = "0.11.0"
-starknet_api = { version = "=0.10", features = ["testing"] }
+starknet_api = { git = "https://github.com/Moonsong-Labs/sequencer", rev = "01b97d9354faefa5f0cd520b5efbe73178213f7e", features = ["testing"] }
```
Comment (HermanObst) [inline Cargo.toml:61]: Are we pulling `starknet_api` from `sequencer`?

Comment (odesenfans) [reply Cargo.toml:61]: Yes, we need to have the exact same version as Blockifier.

File: crates/rpc-replay/src/utils.rs [hunk 2]
```diff
@@ -11,12 +10,12 @@ where
...
-pub fn felt_api2vm(felt: StarkFelt) -> Felt {
-    Felt::from_bytes_be_slice(felt.bytes())
+pub fn felt_api2vm(felt: Felt) -> Felt {
+    felt
 }
 
-pub fn felt_vm2api(felt: Felt) -> StarkFelt {
-    StarkFelt::new_unchecked(felt.to_bytes_be())
+pub fn felt_vm2api(felt: Felt) -> Felt {
+    felt
```
Comment (HermanObst) [inline crates/rpc-replay/src/utils.rs]: Do we need this?

Comment (odesenfans) [reply crates/rpc-replay/src/utils.rs]: I remove it in later commits.

**Comment by odesenfans** [comment]
Replaced by #328 for CI issues.

**Pull request closed without merging**